x/concentrated-liquidity/fees_test.go

```diff
@@ -1015,3 +1020,152 @@ func (s *KeeperTestSuite) TestPreparePositionAccumulator() {
+	ticksActivatedAfterEachSwapUp, totalFeesExpectedUp := s.swapAndTrackXTimesInARow(clPool.GetId(), DefaultCoin1, ETH, cltypes.MaxSpotPrice, 7)
+	ticksActivatedAfterEachSwapDown, totalFeesExpectedDown := s.swapAndTrackXTimesInARow(clPool.GetId(), DefaultCoin0, USDC, cltypes.MinSpotPrice, 7)
+	totalFeesExpected = totalFeesExpectedUp.Add(totalFeesExpectedDown...)
+
+	// We expect all positions to have both denoms in their fee accumulators except USDC for the overlapping range position since
+	// it was not activated during the USDC -> ETH swap direction but was activated during the ETH -> USDC swap direction.
+	ticksActivatedAfterEachSwapTest := [][]sdk.Int{ticksActivatedAfterEachSwapUp, ticksActivatedAfterEachSwapDown}
+	denomsExpected := [][]string{{USDC, ETH}, {USDC, ETH}, {USDC, ETH}, {"", ETH}}
+
+	s.CollectAndAssertFees(s.Ctx, clPool.GetId(), totalFeesExpected, ticksActivatedAfterEachSwapTest, denomsExpected)
+}
+
+// CollectAndAssertFees collects fees from a given pool for all positions and verifies that the total fees collected match the expected total fees.
+// The method also checks that if the ticks that were active during the swap lie within the range of a position, then the position's fee accumulators
+// are not empty. The total fees collected are compared to the expected total fees within an additive tolerance defined by an error tolerance struct.
+func (s *KeeperTestSuite) CollectAndAssertFees(ctx sdk.Context, poolId uint64, totalFees sdk.Coins, activeTicks [][]sdk.Int, expectedFeeDenoms [][]string) {
+	var totalFeesCollected sdk.Coins
+	// Claim full range position fees across all four accounts
+	for i := 0; i < 4; i++ {
+		coins, err := s.App.ConcentratedLiquidityKeeper.CollectFees(ctx, poolId, s.TestAccs[i], DefaultMinTick, DefaultMaxTick)
+		s.Require().NoError(err)
+		totalFeesCollected = totalFeesCollected.Add(coins...)
+		s.tickStatusInvariance(activeTicks, DefaultMinTick, DefaultMaxTick, coins, expectedFeeDenoms[0])
+	}
+
+	// Claim narrow range position fees across three of four accounts
+	for i := 0; i < 3; i++ {
+		coins, err := s.App.ConcentratedLiquidityKeeper.CollectFees(ctx, poolId, s.TestAccs[i], DefaultLowerTick, DefaultUpperTick)
+		s.Require().NoError(err)
+		totalFeesCollected = totalFeesCollected.Add(coins...)
+		s.tickStatusInvariance(activeTicks, DefaultLowerTick, DefaultUpperTick, coins, expectedFeeDenoms[1])
+	}
+
+	// Claim consecutive range position fees across two of four accounts
+	for i := 0; i < 2; i++ {
+		coins, err := s.App.ConcentratedLiquidityKeeper.CollectFees(ctx, poolId, s.TestAccs[i], DefaultExponentConsecutivePositionLowerTick.Int64(), DefaultExponentConsecutivePositionUpperTick.Int64())
+		s.Require().NoError(err)
+		totalFeesCollected = totalFeesCollected.Add(coins...)
+		s.tickStatusInvariance(activeTicks, DefaultExponentConsecutivePositionLowerTick.Int64(), DefaultExponentConsecutivePositionUpperTick.Int64(), coins, expectedFeeDenoms[2])
+	}
```

Comment:
nit: I think this loop can be extracted into a helper and the helper called 3 times instead of repeating a similar loop

Reply:
https://github.com/osmosis-labs/osmosis/pull/4282/commits/28b4ec1e23aebd23c858b6d4db8195cf34394711